File: week-1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main()
{

  int height = 0;
  int space = 0;
  int gap = 2;

  do
  {
    height = get_int("Height: ");
  } while (!(height >= 1 && height <= 8));

  space = height - 1;

  for (int row = 0; row < height; row++)
  {
    // add space before #
    for (int s = 0; s < space; s++)
    {
      printf(" ");
    }

    // print left pyramid
    for (int hash = 0; hash < row + 1; hash++)
    {
      printf("#");
    }

    // add gap size 2 as space before #
    for (int g = 0; g < gap; g++)
    {
      printf(" ");
    }

    // print right pyramid
    for (int hash = 0; hash < row + 1; hash++)
    {
      printf("#");
    }

    space -= 1;
    printf("\n");
  }
}

File: week-6/sentimental-mario-more/mario.py
from cs50 import get_int

# ask user for half pyramid's height until height is between 1 and 8
height = 0
while True:
    try:
        height = get_int("Height: ")
    except ValueError:
        continue

    if height >= 1 and height <= 8:
        break
    else:
        continue

space = height - 1
gap = 2

# loop through
for row in range(height):
    # add space before #
    for _ in range(space):
        print(" ", end="")

    # left pyramid
    for _ in range(row + 1):
        print("#", end="")

    # add gap size 2 as space before #
    for _ in range(gap):
        print(" ", end="")

    # right pyramid
    for _ in range(row + 1):
        print("#", end="")

    space -= 1
    print()
